src/main/groovy/io/github/groovymc/modsdotgroovy/ConvertToQuiltJsonTask.groovy

```diff
@@ -0,0 +1,124 @@
+package io.github.groovymc.modsdotgroovy
+
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.moandjiezana.toml.TomlWriter
+import groovy.transform.CompileDynamic
+import groovy.transform.CompileStatic
+import org.codehaus.groovy.control.CompilerConfiguration
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.CopySpec
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.MapProperty
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.SourceSet
+import org.gradle.api.tasks.TaskAction
+import org.gradle.language.jvm.tasks.ProcessResources
+
+import java.nio.file.Files
+
+abstract class ConvertToQuiltJsonTask extends DefaultTask {
```

Comment:
This seems like a lot of duplicated code from `ConvertToTomlTask`.
Please add an abstract common class which handles the duplicated code.
Additionally, the shell environment should be given the `modLoaderType` as an argument.